doc/neps/nep-0052-python-api-cleanup.rst

```diff
@@ -89,29 +111,32 @@ NumPy major version number.
 Backward compatibility
 ----------------------
 
-There is a backwards compatibility impact for users of deprecated or removed
-functionality, as well as for users and libraries that were relying on private
-NumPy APIs that will move due to the clearer namespacing.
+As mentioned above, while the new API should be backward compatible,
+there is no guarantee of forward compatibility.
+Code will have to be updated to account for deprecated, moved, or
+removed functions/classes, as well as for more strictly enforced private APIs.
 
-In order to make it easier to adapt to the changes in this NEP, we will:
+In order to make it easier to adopt to the changes in this NEP, we will:
 
-1. Ensure that NumPy 2.0 will come with guidance for each removed API, pointing
-   to the preferred alternative API to use,
+1. Provide a transition guide that lists each API change and its replacement.
 2. Provide a script to automate the migration wherever possible. This will be
    similar to ``tools/replace_old_macros.sed`` (which adapts code for a
    previous C API naming scheme change).
 
+.. S: Q: is (2) too ambitious? I don't know how easy it is to do this
+   under all circumstances.
```

Comment:
I think "wherever possible" covers that - basically all aliases and simple stuff. I'm thinking to do what can be done with `sed` or a similar tool, nothing like AST analysis.